### PR TITLE
Igonchar/tdi 39271 remove salesforce edit wizard

### PR DESCRIPTION
**Pull request by gonchar-ivan**
**What is the current behavior?** (You can also link to an open issue here)

Salesforce has 2 connection wizard definition. One of them is redundant 

**What is the new behavior?**

Salesforce has 1 connection wizard definition

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://jira.talendforge.org/browse/TDI-39271

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/901/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/901/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/902/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/902/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/905/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/905/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/906/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/906/)

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java [hunk 1]
```diff
@@ -300,10 +297,10 @@ public void testWizard() throws Throwable {
 
         SalesforceConnectionProperties connProps = (SalesforceConnectionProperties) connFormWizard.getProperties();
 
-        Form af = connProps.getForm(Form.ADVANCED);
+        Form advancedForm = connProps.getForm(Form.ADVANCED);
         assertTrue(
```
Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java]: As we discussed earlier, it's better to use **assertEquals**.

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java [hunk 2]
```diff
@@ -375,34 +372,21 @@ public void testWizard() throws Throwable {
...
                 .toArray(new ComponentWizard[2]);
         Arrays.sort(subWizards, new WizardNameComparator());
         assertEquals(2, subWizards.length);
-        // Edit connection wizard - we copy the connection properties, as we present the UI, so we use the
-        // connection properties object created by the new wizard
-        assertFalse(connProps == subWizards[1].getForms().get(0).getProperties());
+
         assertFalse(subWizards[1].getDefinition().isTopLevel());
-        assertEquals("Edit Salesforce Connection", subWizards[1].getDefinition().getMenuItemName());
+        assertEquals("Salesforce Modules", subWizards[1].getDefinition().getMenuItemName());
```
Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java]: I think you can reorder **subWizards[0]** and **subWizards[1]** asserts.

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java (hunk 1 above)
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java]: I assume here the author checks whether it is the same instance rather then 2 instances are equal, so I'll leave it as is

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java (hunk 2 above)
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/integration/SalesforceComponentTestIT.java]: thanks. will do

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/939/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/939/)

**Pull request merged**